src/Hpack/Config.hs

```diff
@@ -1276,6 +1288,15 @@ toSection = go
       where
         conditional (Condition (Cond c)) = Conditional c
 
+toBuildTool :: String -> [String] -> ParseBuildTool -> BuildTool
+toBuildTool packageName_ executableNames = \ case
+  QualifiedBuildTool pkg executable
+    | pkg == packageName_ && executable `elem` executableNames -> LocalBuildTool executable
```

Comment:
This might go wrong if people define an executable in a verbatim section, but I guess that using verbatim means that you know what you're doing. Possibly worth calling out this case in the docs, as a hint for why hpack would be doing the 'wrong' thing.

Reply:
Regarding docs, how about https://github.com/sol/hpack/pull/323?